test/statsig/bootstrap_test.exs

```diff
@@ -0,0 +1,46 @@
+defmodule Statsig.BootstrapTest do
+  use ExUnit.Case
+
+  @moduletag :capture_log
+
+  setup do
+    # Stop the application if it's running
+    Application.stop(:statsig)
+
+    # Read and set bootstrap data
+    bootstrap_data =
+      "test/data/rulesets_e2e_config.json"
+      |> File.read!()
+      |> Jason.decode!()
+
+    Application.put_env(:statsig, :bootstrap_config_specs, bootstrap_data)
+
+    # Now start the application with bootstrap data in place
+    start_supervised!(%{
+      id: Statsig.Application,
+      start: {Statsig.Application, :start, [:normal, []]}
+    })
+
+    on_exit(fn ->
+      Application.delete_env(:statsig, :bootstrap_config_specs)
+    end)
+
+    :ok
+  end
+
+  test "uses bootstrapped gate values" do
+    {:ok, result} = Statsig.check_gate(%Statsig.User{user_id: "12345"}, "test_numeric_user_id")
```

Comment:
for gates, there's a three way logic here that I'm not sure you intended. 
One option is some kind of error occurring, the other is that the result is true or false. 

Can you describe how checking a gate can fail? As a user, I'd prefer:
```elixir
if Statsig.check_gate?(...) do 
  ...
 end
 ```